core/services/inventory.py

```diff
@@ -184,7 +184,8 @@ def createMultipleTransactionType(self,transactionTypes) -> None:
         return
 
     def updateInventoryItem(self,id:int,inventoryItem) -> None:
-        if self.inventory_repo.getInventoryItemBySalesServiceItemId(inventoryItem.sales_service_item_id): 
+        if self.inventory_repo.getInventoryItemBySalesServiceItemId(inventoryItem.sales_service_item_id) and id != self.inventory_repo.getInventoryItemBySalesServiceItemId(inventoryItem.sales_service_item_id).id:
```

Comment:
bro query this only once.

`ext_inventoryItem = self.inventory_repo.getInventoryItemBySalesServiceItemId(inventoryItem.sales_service_item_id)
if ...
`
And I am not sure, but have you considered what if the function doesn't return anything. Will the second condition case problem since it's going to be accessing null value.

Reply:
yeah, it will be fine. if database return null, id will compare null value and the condiition will be true.